src/modules/cpu/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,106 @@
+#include "rppdefs.h"
+
+Rpp32f getSquare(Rpp32f &value) 
+{
+  Rpp32f res = value;
+  return (res * res);
+}
+
+Rpp32f getMax(std::vector<float> &values, int length)
+{
+  Rpp32f max = values[0];
+  for (int i = 1; i < length; i++) 
+  {
+    max = std::max(max, values[i]);
+  }
+  return max;
+}
+
+RppStatus non_silent_region_detection_host_tensor(Rpp32f *srcPtr,
+                                                  RpptDescPtr srcDescPtr,
+                                                  Rpp32s *srcSizeTensor,
+                                                  Rpp32s *detectedIndexTensor,
+                                                  Rpp32s *detectionLengthTensor,
+                                                  Rpp32f *cutOffDBTensor,
+                                                  Rpp32s *windowLengthTensor,
+                                                  Rpp32f *referencePowerTensor,
+                                                  Rpp32s *resetIntervalTensor,
+                                                  bool *referenceMaxTensor)
+{
+#pragma omp parallel for num_threads(srcDescPtr->n)    
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+      Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->w;
```

Comment:
srcDescPtr->w and srcDescPtr->strides.nStride should ideally be the same for audio correct? Since there is no height and no channels?
Using srcDescPtr->strides.nStride conveys better as to how much stride we are jumping.